src/Build/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs

```diff
@@ -1322,9 +1322,9 @@ private int GetNextBuildRequestId()
         /// <param name="config">The configuration to be mapped.</param>
         private void IssueConfigurationRequest(BuildRequestConfiguration config)
         {
-            ErrorUtilities.VerifyThrowArgument(config.WasGeneratedByNode, "InvalidConfigurationId");
```

Comment:
> it does change the type of exception thrown if it fails

I'm not sure it can ever have thrown ArgumentException. Before this change it presumably threw MissingManifestResourceException. (My guess is that it's never thrown at all but I guess we could introduce a bug that would trigger it)

Reply:
Oh, because InvalidConfigurationId doesn't exist? I'm kinda wondering if we should have an explicit part of CI that prevents you from checking in a use of a localized string without actually adding that string...

Reply:
@Forgind exactly. the usual way to handle this is to use strongly typed resource strings. but that would not solve the problem here because the VerifyThrow things (that throw InternalErrorException) use unlocalized strings. So they'd still get raw strings and the two could still get mixed up. Hence my suggestion in https://github.com/dotnet/msbuild/issues/8991